src/main/java/com/cinderella/dao/mysql/MySQLConnection.java

```diff
@@ -180,8 +176,57 @@ public WashMachine findWashMachineById(long machineId) {
 	 */
 	@Override
 	public boolean updateUser(User user) {
-
+		if (conn == null) {
+			return false;
+		}
+		try {
+			String sql = "INSERT IGNORE INTO user("
```

Comment:
It's better move the user creation into another method called "addUser(User user)" to decouple the different use cases and make the Dao easier to be used by the clients/callers.